chia/full_node/mempool.py

```diff
@@ -37,37 +37,37 @@ def get_min_fee_rate(self, cost: int) -> float:
         else:
             return 0
 
-    def remove_from_pool(self, item: MempoolItem):
+    def remove_from_pool(self, items: List[bytes32]) -> None:
         """
         Removes an item from the mempool.
         """
-        removals: List[Coin] = item.removals
-        additions: List[Coin] = item.additions
-        for rem in removals:
-            del self.removals[rem.name()]
-        for add in additions:
-            del self.additions[add.name()]
-        del self.spends[item.name]
-        del self.sorted_spends[item.fee_per_cost][item.name]
-        dic = self.sorted_spends[item.fee_per_cost]
-        if len(dic.values()) == 0:
-            del self.sorted_spends[item.fee_per_cost]
-        self.total_mempool_cost -= item.cost
-        assert self.total_mempool_cost >= 0
-
-    def add_to_pool(
-        self,
-        item: MempoolItem,
-    ):
+        for spend_bundle_id in items:
+            item = self.spends[spend_bundle_id]
+            assert item.name == spend_bundle_id
+            removals: List[Coin] = item.removals
+            for rem in removals:
+                rem_name: bytes32 = rem.name()
+                self.removals[rem_name].remove(spend_bundle_id)
+                if len(self.removals[rem_name]) == 0:
+                    del self.removals[rem_name]
+            del self.spends[item.name]
+            del self.sorted_spends[item.fee_per_cost][item.name]
+            dic = self.sorted_spends[item.fee_per_cost]
+            if len(dic.values()) == 0:
+                del self.sorted_spends[item.fee_per_cost]
+            self.total_mempool_cost -= item.cost
+            assert self.total_mempool_cost >= 0
+
+    def add_to_pool(self, item: MempoolItem) -> None:
         """
         Adds an item to the mempool by kicking out transactions (if it doesn't fit), in order of increasing fee per cost
         """
 
         while self.at_full_capacity(item.cost):
             # Val is Dict[hash, MempoolItem]
             fee_per_cost, val = self.sorted_spends.peekitem(index=0)
-            to_remove = list(val.values())[0]
-            self.remove_from_pool(to_remove)
+            to_remove: MempoolItem = list(val.values())[0]
+            self.remove_from_pool([to_remove.name])
```

Comment:
so we kick out transactions without checking that the new `MempoolItem` has a higher fee_per_cost?

Reply:
That was checked in validation